File: ko2cogc.rb
#!/usr/bin/env ruby

strID = nil
hashMap = {}
STDIN.each do |strLine|
	if( strLine =~ /^ENTRY\s+(\S+)/ )
		strID = $1
	elsif( strLine =~ /^(?:DBLINKS)?\s*COG:(.+)$/ )
		hashMap[strID] = $1.strip.split( /\s+/ ); end; end
puts( hashMap.map do |strKO, astrCOGs|
	( [strKO] + astrCOGs ).join( "\t" ); end )

File: src/ko2cogc.py
#!/usr/bin/env python

import re
import sys

strKO = None
hashCOGs = {}
for strLine in sys.stdin:
	mtch = re.search( '^ENTRY\s+(K\d+)', strLine )
	if mtch:
		strKO = mtch.group( 1 )
		continue
	mtch = re.search( '^(?:DBLINKS)?\s*COG:(.+)$', strLine )
	if mtch:
		hashCOGs.setdefault( strKO, [] ).extend( mtch.group( 1 ).strip( ).split( " " ) )

for strKO, astrCOGs in hashCOGs.items( ):
	print( "\t".join( [strKO] + astrCOGs ) )
